packages/core/utils/src/config.js

```diff
@@ -32,15 +35,18 @@ export async function resolveConfig(
 
   for (const filename of filenames) {
     let file = path.join(filepath, filename);
-    if ((await fs.exists(file)) && (await fs.stat(file)).isFile()) {
-      return file;
+    try {
+      if ((await fs.stat(file)).isFile()) {
```

Comment:
You could also cache whether a particular file exists in a separate map (e.g. Map<FilePath, boolean>). This way we don't have to keep searching the filesystem even when we already have the resolved content cached.